### PR TITLE
[TPSVC-17155] Added support for custom ECS fields

### PR DESCRIPTION
**Pull request by IevgenTarasov**
**What is the problem this Pull Request is trying to solve?**
 https://jira.talendforge.org/browse/TPSVC-17155

**What is the chosen solution to this problem?**
 
**Link to the JIRA issue**

 
**Please check if the Pull Request fulfills these requirements**
- [ ] The PR commit message follows our [guidelines](https://github.com/Talend/daikon/blob/master/CONTRIBUTING.md)
- [ ] Tests for the changes have been added (for bug fixes / features, coverage should be over 75% in the new code)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Related design / discussions / pages (not in JIRA), if any, are all linked or available in the Pull Request

 
**[ ] This Pull Request introduces a breaking change**

### CODE REVIEW

**Review by jehervy — APPROVED** [review]
LGTM

File: daikon-logging/logging-event-layout/src/main/resources/custom_flat.yml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+span.parent.id:
```
Comment (sgandon) [inline daikon-logging/logging-event-layout/src/main/resources/custom_flat.yml:1]:
```suggestion
#List of authorized custome field for Talend applications.
# make sure that are all in lower case and of course not already part of the official ECS fields.
span.parent.id:
```

Comment (sgandon) [reply daikon-logging/logging-event-layout/src/main/resources/custom_flat.yml:1]: added some suggested comments

File: daikon-logging/logging-event-layout/src/main/java/org/talend/daikon/logging/event/layout/LogbackJSONLayout.java [hunk 2]
```diff
@@ -31,7 +30,7 @@
 
     private String serviceName;
 
-    private final List<AdditionalField> additionalFields = new ArrayList<AdditionalField>();
+    private final List<AdditionalField> additionalFields = new ArrayList<>();
```
Comment (sgandon) [inline daikon-logging/logging-event-layout/src/main/java/org/talend/daikon/logging/event/layout/LogbackJSONLayout.java:33]: what is the purpose of this field ? it is private and does not seem to be used in this class.
Am I missing something ?

**Pull request merged**